src/App.tsx

```diff
@@ -58,203 +57,99 @@ const PageNumber = styled.div`
-  }, [update, selectedID]);
+  }, [state.search, state.page, state.mediaType]);
 
   const sortedMovies = React.useMemo(() => {
-    return state.result
-      ? state.result.Search.slice(0).sort((a, b) => {
+    return state.movies
+      ? state.movies.slice(0).sort((a, b) => {
           return parseInt(a.imdbRating) - parseInt(b.imdbRating);
         })
       : [];
-  }, [state.result?.Search]);
+  }, [state.movies]);
 
-  return !state.result ? null : (
+  return !state.movies ? null : (
     <MainContainer>
       <Title>Movie Database</Title>
-      {popUpState ? (
+      {state.popupState && state.selectedMovie && state.movies.length ? (
         <PopUp
-          selected={state.selectedMovie}
-          backClick={(click) => {
-            updatePopUpState(click);
-            update({
-              type: "select movie response set",
-              selMovieResPayload: initialDetailedState,
-            });
-            updateSelectedID("");
-          }}
+          selected={state.movies.find(
+            (movie) => movie.imdbID === state.selectedMovie
+          )}
+          backClick={() =>
+            void update({
+              type: "popup back button clicked",
+            })
+          }
         />
       ) : (
         <div>
           <SearchForm
-            submit={({ search, mediaType }) => {
-              update({ type: "search set", payload: { search } });
-              update({ type: "page number changed", pagePayload: { page: 1 } });
-              setMediaType(mediaType);
-            }}
+            submit={({ search, mediaType }) =>
+              void update({
+                type: "search submitted",
+                payload: {
+                  mediaType,
+                  search,
+                },
+              })
+            }
           />
-
           <ResultsFound>
-            {state.result.totalResults} results for "{state.search}"
+            {state.movies.length} results for "{state.search}"
           </ResultsFound>
 
           <MovieList
-            movies={state.result.Search}
-            sendSelectedID={(id) => {
-              updateSelectedID(id);
-              updatePopUpState(true);
-            }}
+            movies={state.movies}
+            clicked={(id) =>
+              void update({
+                type: "movies list item clicked",
+                clickedMovieId: id,
+              })
+            }
           />
 
           <HighRatedList highRatedMovies={sortedMovies} />
           <NavButtons>
             {state.page === 1 ? null : (
               <NavButton
-                onClick={() => {
-                  update({
-                    type: "page number changed",
-                    pagePayload: { page: state.page - 1 },
-                  });
-                }}
+                onClick={() =>
+                  void update({
+                    type: "page button minus clicked",
+                  })
+                }
               >
```

Comment:
With the page button clicks, is it beneficial to pull the 'logic' into the switch cases rather than having a payload which i updated on this side?

Reply:
yeah it keeps the presentation simpler by pulling the "business logic" into the reducer

Reply:
Are there any cases where you wouldn't put the logic in the reducer, or should a keep it black and white?

Reply:
Potentially if you have a mixture of state if you had some state local to a component you won't be able to access that in the reducer so you may decide to compute the next state in the event handler for example. However, alternatively in this situation you could just pass the external state to the reducer in the event payload.